e2e/test/AppContextTests.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    [TestCategory("AppContext")]
+    public class AppContextTests
+    {
+        private readonly string DevicePrefix = $"E2E_{nameof(AppContextTests)}_";
+        private static TestLogging _log = TestLogging.GetInstance();
+
+        private readonly ConsoleEventListener _listener;
+
+        public AppContextTests()
+        {
+#if !NET451
+            AppContext.SetSwitch(AppContextConstants.DisableObjectDisposedExceptionForReceiveAsync, true);
+#endif
+
+            _listener = TestConfig.StartEventListener();
+        }
+
+        #region ReceiveAsyncAfterDispose
+
+        [TestMethod]
+        public async Task AppConfig_ReceiveAsyncAfterDispose_Sasl_Amqp()
+        {
+
+            bool isFlagSet = false;
+            bool flag = false;
+
+#if !NET451
+            isFlagSet = AppContext.TryGetSwitch(AppContextConstants.DisableObjectDisposedExceptionForReceiveAsync, out flag);
+#endif
+
+            if (isFlagSet)
+            {
+                await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
```

Comment:
Do not use acronyms.

Reply:
The "DCLC_" has been removed everywhere in the file.

Reply:
And also in DEviceClientLifeCycleTests.cs